messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,145 @@
+= [WFLY-13793] Allow for a remote jms queue / topic not to use legacy amq1 prefix
+:author:            Chao Wang
+:email:             chaowan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+Currently we can disable the use of legacy amq1 prefixes `jms.queue` and `jms.topic` only at the connection factory level but not for queue or topic defined in the model or using JakartaEE annotations.
+The goal of this proposal is to be able to disable the usage of such prefixes at the destination level for external JMS queues and topics. Thus we would avoid creation of prefixed destinations that will be useless.
+We start by following the procedure described in https://github.com/wildfly/quickstart/tree/main/remote-helloworld-mdb[Remote MDB Quickstart]. Basically what's relevant here is:
+* we start a remote Artemis Broker which is NOT running in compatibility mode (= it doesn't try to add prefixes for all incoming queue/topic names)
+* we create a connection on WildFly to the remote Artemis Broker (called pooled-connection-factory) which overrides the default value of enable-amq1-prefix (which is true) by setting it to enable-amq1-prefix=false: this means also the WildFly Artemis client won't try to add prefixes to all queue/topic names
+* we create an application which defines a queue named HelloWorldMDBQueue which mimics the default behavior by setting *properties = {"enable-amq1-prefix=true"}* (default would be true) on the queue definition:
+
+[source, java]
+----
+@JMSDestinationDefinition(
+    name = "java:/queue/HELLOWORLDMDBQueue",
+    interfaceName = "jakarta.jms.Queue",
+    destinationName = "HelloWorldMDBQueue"
+)
```

Comment:
@soul2zimate , @ehsavoie  since we explicitly mentioned it, shouldn't we add `properties = {"enable-amq1-prefix=true"}` here even if it is the default?

Reply:
Yes, it's fine to explicitly add it. 
@ehsavoie May I get your blessing on this as well ?

Reply:
you have :anatomical_heart:

Reply:
Updated, Once this is approved. I will squash commits.